test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -459,37 +451,29 @@ def load_hook(self, filename):
                 script = fd.read()
         except:
             self.assertTrue(False, "Failed to open hook file %s" % filename)
-        newfile = os.path.join(os.sep, 'tmp', self.random_name())
-        with open(newfile, "w") as fd:
-            fd.write(script)
-        try:
-            events = '"execjob_begin,execjob_launch,execjob_attach,'
-            events += 'execjob_epilogue,execjob_end,exechost_startup,'
-            events += 'exechost_periodic"'
-            a = {'enabled': 'True',
-                 'freq': '2',
-                 'event': events}
-            self.server.create_import_hook(self.hook_name, a, script,
-                                           overwrite=True)
-            # Add the configuration
-            self.load_config(self.cfg0)
-        except:
-            raise
-        finally:
-            os.remove(newfile)
+        events = '"execjob_begin,execjob_launch,execjob_attach,'
+        events += 'execjob_epilogue,execjob_end,exechost_startup,'
+        events += 'exechost_periodic"'
+        a = {'enabled': 'True',
+             'freq': '2',
+             'event': events}
+        self.server.create_import_hook(self.hook_name, a, script,
+                                       overwrite=True)
+        # Add the configuration
+        self.load_config(self.cfg0)
 
     def load_config(self, cfg):
         """
         Create a hook configuration file with the provided contents.
         """
-        cfg_file = os.path.join(os.sep, 'tmp', self.random_name())
-        with open(cfg_file, "w") as fd:
-            fd.write(cfg)
+        (fd, fn) = tempfile.mkstemp()
```

Comment:
Can you please use self.du.mkstemp() instead of tempfile.mkstemp()

Reply:
Hey Arun, I looked up at the PTL documentation and they say following

Create a temp file by calling tempfile.mkstemp

Reply:
That is interesting... I saw PTL framework itself using disk utility for mkstemp. I'd like to hear from @hirenvadalia if this is what we should be doing while creating a file.

Reply:
@anamikau I think @arungrover is right... we should self.du.mkstemp() whenever it is possible...
Regarding doc: I think that's not right! if doc is saying to use tempfile.mkstemp over self.du.mkstemp. we should change doc for same. Please file ticket and possible please change it 😄

Reply:
Done.